src/components/main/settings/FilterMembers.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable react/prop-types */
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { List } from 'immutable';
+import { AppSetting, Member } from '@graasp/apps-query-client';
+import { Autocomplete, Checkbox, TextField } from '@mui/material';
+import {
+  PatchAppSettingType,
+  PostAppSettingType,
+} from '../../../types/appSettings';
+import { APP_SETTINGS_TYPES } from '../../../config/appSettingTypes';
+
+interface FilterMembersProps {
+  members: List<Member>;
+  settings: List<AppSetting>;
+  postAppSetting: (s: PostAppSettingType) => void;
+  patchAppSetting: (s: PatchAppSettingType) => void;
+}
+
+type FilteredMembersSettingType = AppSetting & {
+  data: {
+    filteredMembers: Array<string>;
+  };
+};
+
+const FilterMembers = (fcProps: FilterMembersProps): JSX.Element => {
+  const { t } = useTranslation();
+  const { members, postAppSetting, patchAppSetting, settings } = fcProps;
+
+  const [filteredMembers, setFilteredMembers] = useState<Member[]>([]);
+  const [settingId, setSettingId] = useState<string>();
+
+  useEffect(() => {
+    const filteredMembersSetting: FilteredMembersSettingType | undefined =
+      settings.find(
+        (s) => s.name === APP_SETTINGS_TYPES.FILTERED_MEMBERS,
+      ) as FilteredMembersSettingType;
```

Comment:
Since you have a context for your settings, maybe you can directly extract your settings in the context and provide them directly.

Reply:
Good idea. I'll refactor this in another PR. See #24